### PR TITLE
Change how grades are returned. 

### PR DESCRIPTION
**Pull request by Lathryx**
_Firstly_, `grades` are returned by the IC API using multiple different names, so now this library accounts for that. 

_Secondly_, the function in order to account for courses without grades was changed slightly to work with both variables as well. 

_Thirdly_, this update adds some slight restructuring to ensure users with multiple schools can still use the API (and the `comments` property was added to the returned data as well).

### CODE REVIEW

**Review by qwazwsx — CHANGES_REQUESTED** [review]
Everything looks great! However, I notice a bug in lines 308-311. If any of the `.progress` properties are nullish (i.e. 0) it will trigger the "no grades" if statement and set .grades to undefined. Another reason why nullish coalescing is great. I'll make a PR real quick and fix it.

```Javascript

// the following works
5 || undefined == 5 // !!5 == true, picks the left expression
undefined || 5 == 5 // !!undefined == false, picks the right expression

// here is the error, the || operator casts the values to booleans, and 0 is a falsy value.
0 || undefined == undefined // !!0 == false, picks the right expression


```

**Comment by Lathryx** [comment]
> Everything looks great! However, I notice a bug in lines 308-311. If any of the `.progress` properties are nullish (i.e. 0) it will trigger the "no grades" if statement and set .grades to undefined. Another reason why nullish coalescing is great. I'll make a PR real quick and fix it.
> 
> ```js
> // the following works
> 5 || undefined == 5 // !!5 == true, picks the left expression
> undefined || 5 == 5 // !!undefined == false, picks the right expression
> 
> // here is the error, the || operator casts the values to booleans, and 0 is a falsy value.
> 0 || undefined == undefined // !!0 == false, picks the right expression
> ```
\- @qwazwsx 

Hmm, I could've sworn I fixed this with _this change:_ 
```js
// from this: 
// if (!grade.progressScore 
//        && !grade.progressPercent 
//        && !grade.progressTotalPoints 
//        && !grade.progressPointsEarned) 
//         courseResult.grades = undefined 
// to this: 
  if (!(grade.progressScore || grade.score) 
        && !(grade.progressPercent || grade.percent) 
        && !(grade.progressTotalPoints || grade.totalPoints) 
        && !(grade.progressPointsEarned || grade.pointsEarned))
            courseResult.grades = undefined;
```
This (^) is what you were talking about, correct?

File: index.js [hunk 1]
```diff
@@ -266,17 +295,17 @@ class User extends EventEmitter {
...
+                  pointsEarned: (grade.progressPointsEarned !== undefined) ? grade.progressPointsEarned : grade.pointsEarned
                 },
+                comments: grade.comments,
                 _id: course._id
               }
 
               // remove grades for courses without grades
-              if (!grade.progressScore && !grade.progressPercent && !grade.progressTotalPoints && !grade.progressPointsEarned)
-                courseResult.grades = undefined;
+              if (grades.score === undefined || grades.percent === undefined) courseResult.grades = undefined;
```
Comment (Lathryx) [inline index.js:308]: @qwazwsx I see now that my change (seen above) did _not_ go through after all. 🤔

Comment (qwazwsx) [reply index.js:308]: Hmm... Okay, I see. I completely misread your commit. I thought that the if statement would trigger if *any* of the conditions were met (IE falsy value boolean behavior would apply) but I see now that isn't the case. Sorry about this, I'm re-merging your commit right now. This whole code-review thing is pretty new to me. Egg on my face :grimacing:

**Comment by qwazwsx** [comment]
the documentation still needs updating to reflect the multiple schools change to `User.getCourses`
I also still need to push V3.1.0 to NPM

**Comment by Lathryx** [comment]
Got it. I can help with the docs if you want. I'm eagerly waiting for the NPM update (currently working on my project's back-end, where I'll be using this API)! 

Also, send me a friend request on Discord _(Lathryx#2826)_ when you get a chance! :)

File: index.js (hunk 1 above)
Comment (Lathryx) [reply index.js:308]: Haha! No worries at all. I'm pretty new to all this collaborative stuff too. :)

**Pull request merged**